component/constraint/CableConstraint.inl

```diff
@@ -63,9 +63,7 @@ CableConstraint<DataTypes>::CableConstraint(MechanicalState* object)
                                           "force = the contstraint will impose the force provided in data d_inputValue[d_iputIndex] \n"
                                           "If unspecified, the default value is displacement"))
 {
-    d_value.setGroup("Input");
-    d_valueIndex.setGroup("Input");
-    d_valueType.setGroup("Input");
+    setUpData();
```

Comment:
It is good to factor things.
I think you should go one step further by using constructor delegate
https://www.geeksforgeeks.org/constructor-delegation-c/
(using them you don't need to duplicate the "initData" part).

Reply:
I think I cannot because of the current delegation: 

```
CableConstraint<DataTypes>::CableConstraint(MechanicalState* object)
    : Inherit(object)
...

template<class DataTypes>
CableConstraint<DataTypes>::CableConstraint()
    : Inherit()
...
```

if I do something like: 

```
CableConstraint<DataTypes>::CableConstraint(MechanicalState* object)
    : Inherit(object)
     , CableConstraint<DataTypes>()
...
``` 

there will be a conflict with Inherit() and Inherit(object). I cannot think of a way to bypass that.